products/platform/documentation-site/research/search-design-doc.md

```diff
@@ -1,15 +1,114 @@
+
+Algolia appears to be the leader when it comes to searching documentation.
+
+### Code Location
+
+_The path of the source code in the repository._
+
+### Testing Plan
+
+_How you will verify the behavior of your system. Once the system is written, this section should be updated to reflect the current state of testing and future aspirations._
+
+#### Testing the search input
+
+#### Testing the crawler
+
+### Logging
+
+By logging the term(s) that users are querying, we could theoretically identify trends to help increase the quality of our search results.
+
+### Debugging
+
+_How users can debug interactions with your system. When designing a system it's important to think about what tools you can provide to make debugging problems easier. Sometimes it's unclear whether the problem is in your system at all, so a mechanism for isolating a particular interaction and examining it to see if your system behaved as expected is very valuable. Once a system is in use, this is a great place to put tips and recipes for debugging. If this section grows too large, the mechanisms can be summarized here and individual tips can be moved to another document._
+
+### Caveats
+
+- Access Control: By making the documentation landing page public, it becomes difficult, if not impossible, to allow an unauthenticated user to search documentation from private sources (like a private Github repo). One suggestion is that search results lead to public documentation sources, and we include links to private documentation in the public sources. That way, we can rely on existing access control solutions, while still providing a decent solution. One of the major downsides of that approach is auditing, adding, updating, and maintaining links to/from the public and private sources.
+
+### Security Concerns
+
+The search input would need to be sanitized.
+
+If the search input dispatched a request directly to the Algolia API, then there'd be minimal risk of the documentation search negatively affecting other systems. If the request was proxied through our API, then we should be able to rely on any existing DOS protections.
+
+### Privacy Concerns
+
+Since the initial implementation will be an unauthenticated landing page, we will have little, if any, data about the user.
+
+While there shouldn't be any PII in our public documentation sources, it is possible it exists, and hasn't been discovered yet. Therefore, it is possible that our search results could lead public users to sources that contain PII.
+
+### Open Questions and Risks
+
+- Should we index private documentation sources? If so, what private documentation sources can we expose to public users, if any?
+- Should the landing page be public only, or should it also support authenticated users?
+- How often should we reindex our documentation sources? For perspective, Algolia's open source crawler runs every 24 hours.
```

Comment:
Maybe not so pie-in-the-sky if we set GitHub actions to trigger an endpoint on the crawler. 😁

Reply:
Oooh, I like that. Might get weird if we have a source of documentation that _isn't_ GitHub, but I don't think that's the case now. There are also performance considerations, but...implementation details. :smile:

Reply:
We will be wanting to index results from design.va.gov in the MVP, but not sure if that still counts as GitHub (since it’s coming from https://github.com/department-of-veterans-affairs/vets-design-system-documentation and maybe we could use that to trigger the crawler).